### PR TITLE
Preparing the code for further development of CBnT PCR1 calculation support.

### PR DESCRIPTION
**Pull request by xaionaro**
Please review this PR on per-commit basis (for easier review).

### CODE REVIEW

**Review by ChriMarMe — APPROVED** [review]

**Review by walterchris — APPROVED** [review]

File: pkg/pcr/get_measurements.go [hunk 1]
```diff
@@ -53,3 +52,133 @@ func GetMeasurements(
...
+	resultConfig := config
+	debugInfo := map[string]interface{}{
+		"config_orig":   config,
+		"config_result": &resultConfig,
+	}
+
+	// Detect attestation flow
+	detectedFlow, detectFlowErr := DetectAttestationFlow(firmware, config.Registers, config.TPMDevice)
+	if detectedFlow != FlowAuto {
+		debugInfo["detectedAttestationFlow"] = detectedFlow.String()
```
Comment (rihter007) [inline pkg/pcr/get_measurements.go:75]: nit
As we are doing a little bit of refactor, I would also rethink debugInfo. Make all potential keys as public constants should be a good start

File: pkg/pcr/get_measurements.go [hunk 2]
```diff
@@ -53,3 +52,133 @@ func GetMeasurements(
...
+}
+
+func (c *measurementsCollector) FITEntries() []fit.Entry {
+	if c.fitEntriesResult != nil {
+		return *c.fitEntriesResult
+	}
+
+	fitEntries, err := c.firmware.GetFIT()
+	if err != nil {
+		_ = c.errors.Add(ErrGetFIT{Err: err})
```
Comment (rihter007) [inline pkg/pcr/get_measurements.go:122]: nit

make it as a method of measurementsCollector:

func (c *measurementsCollector) addError(err error)

File: pkg/pcr/measurement.go [hunk 3]
```diff
@@ -376,6 +376,26 @@ func (s Measurements) CompileMeasurableData(image []byte) []byte {
 	return result
 }
 
+// FilterByPCRIndex returns a subset of measurements only which corresponds to specified PCR index.
+func (s Measurements) FilterByPCRIndex(pcrIndex ID) Measurements {
+	var r Measurements
+	for _, m := range s {
+		found := false
```
Comment (rihter007) [inline pkg/pcr/measurement.go:383]: nit

var found bool

**Review by rihter007 — APPROVED** [review]
Minor nits, GTG!

File: pkg/pcr/get_measurements.go (hunk 2 above)
Comment (xaionaro) [reply pkg/pcr/get_measurements.go:122]: Yeah. Thanks for the advice. I'll continue changing this code in another PR, hoping won't forget to fix this as well :)

**Pull request merged**